paddle/fluid/framework/ngraph_operator.cc

```diff
@@ -215,6 +271,297 @@ void FusedOperator::RunImpl(const Scope& scope,
   ngraph_op.Run(scope, place);
 }
 
+std::unordered_map<std::string, std::shared_ptr<ngraph::Function>>
+    NgraphOperator::func_cache_ = {};
+
+std::shared_ptr<ngraph::runtime::Backend> NgraphOperator::backend_ =
+    ngraph::runtime::Backend::create("CPU");
+
+void NgraphOperator::GetNgInputShape(std::shared_ptr<OperatorBase> op) {
+  RuntimeInferShapeContext infer_shape_ctx(*op, scope_);
```

Comment:
If NgraphOperator works like a normal operator, you don't need to expose RuntimeInferShapeContext and call it here?

Reply:
Let me look into getting an infershape for ngraph_operator itself.

Reply:
Making a paddle-like ngraph op is what we will do next, but it may not resolve the exposing class RuntimeInferShapeContext issue. The nGraph needs to know the tensor shape  to define the output tensor, so it requires a lazy evaluation by running infershape for each op w/o running the op computation. An infershape api for ngraph_op will likely be a loop of op->InferShape.

To avoid exposing the class, an api RunInferShape was added to run the op->InferShape only. Can this be a workaround to avoid the exposure?
https://github.com/PaddlePaddle/Paddle/blob/bf624ed31b5699943836e503733639bc6e29fc8e/paddle/fluid/framework/operator.cc#L702